src/methods/properties.js

```diff
@@ -501,6 +501,12 @@ export class PropertiesImplementation {
       let key = InternalGetPropertiesKey(P);
       let map = InternalGetPropertiesMap(O, P);
       let propertyBinding = map.get(key);
+      if (propertyBinding === undefined && O.isPartialObject()) {
```

Comment:
I find this a bit surprising. The invariant below (which ought to have been commented) basically states that since O.$GetOwnProperty (above in line 479) returned a descriptor, there must have been a property binding holding the descriptor and hence map.get(key) should always return that descriptor.

Reply:
Can you look more into this, it seems there might be a bug in another part of the code base if this is the case?

Reply:
It seems that $GetOwnProperty nowadays will return a descriptor when there is no binding. That is not necessarily a good thing. I'll look into it some more.

Reply:
Looking at the overall logic: We only get here for simple partial objects. For such objects, we are happy for you to read and write properties even though we are not sure they exist. It seems inconsistent to fault when you try and delete such a property. So, rather than give an error message here, why not use emitPropertyDelete?